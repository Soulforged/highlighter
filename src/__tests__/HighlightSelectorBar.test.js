import React from 'react';
import {
  render,
  fireEvent,
  cleanup,
  waitForElement
} from "@testing-library/react";
import {
  HighlightSelectorBar
} from '../components';

describe("highlight selector bar tests", () => {

  afterEach(cleanup);

  it("attempts to change selected color on option selected", () => {
    const setHighlightColor = jest.fn();
    const { container, getByTestId } = render((
      <HighlightSelectorBar
        highlightColor="red"
        setHighlightColor={setHighlightColor}
      />
    ));
    const newOption = getByTestId("yellow");
    fireEvent.click(newOption);
    expect(setHighlightColor).toHaveBeenCalledWith("yellow");
    expect(container).toMatchSnapshot();
  });

  it("attempts to clear highlights", () => {
    const clearHighlights = jest.fn();
    const { container, getByTestId } = render((
      <HighlightSelectorBar
        highlightColor="red"
        clearHighlights={clearHighlights}
      />
    ));
    const clearOption = getByTestId("clearHighlightsBtn");
    fireEvent.click(clearOption);
    expect(clearHighlights).toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  });
});
