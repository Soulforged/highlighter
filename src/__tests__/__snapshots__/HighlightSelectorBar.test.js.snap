// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highlight selector bar tests attempts to change selected color on option selected 1`] = `
<div>
  <div
    style="flex-direction: row; padding: 10px;"
  >
    <button
      data-testid="red"
      style="width: 50px; height: 32px; margin: 5px; background-color: red; border: 3px solid blue;"
    />
    <button
      data-testid="yellow"
      style="width: 50px; height: 32px; margin: 5px; background-color: yellow; border: 3px solid black;"
    />
    <button
      data-testid="green"
      style="width: 50px; height: 32px; margin: 5px; background-color: green; border: 3px solid black;"
    />
    <button
      data-testid="clearHighlightsBtn"
      style="width: 50px; height: 32px; margin: 5px;"
      title="clear"
      type="button"
    >
      Clear
    </button>
  </div>
</div>
`;

exports[`highlight selector bar tests attempts to clear highlights 1`] = `
<div>
  <div
    style="flex-direction: row; padding: 10px;"
  >
    <button
      data-testid="red"
      style="width: 50px; height: 32px; margin: 5px; background-color: red; border: 3px solid blue;"
    />
    <button
      data-testid="yellow"
      style="width: 50px; height: 32px; margin: 5px; background-color: yellow; border: 3px solid black;"
    />
    <button
      data-testid="green"
      style="width: 50px; height: 32px; margin: 5px; background-color: green; border: 3px solid black;"
    />
    <button
      data-testid="clearHighlightsBtn"
      style="width: 50px; height: 32px; margin: 5px;"
      title="clear"
      type="button"
    >
      Clear
    </button>
  </div>
</div>
`;
